pandas/io/sql.py

```diff
@@ -974,10 +974,16 @@ def insert_data(self) -> tuple[list[str], list[np.ndarray]]:
         for i, (_, ser) in enumerate(temp.items()):
             if ser.dtype.kind == "M":
                 if isinstance(ser._values, ArrowExtensionArray):
-                    with warnings.catch_warnings():
-                        warnings.filterwarnings("ignore", category=FutureWarning)
-                        # GH#52459 to_pydatetime will return Index[object]
-                        d = np.asarray(ser.dt.to_pydatetime(), dtype=object)
+                    import pyarrow as pa
+
+                    if pa.types.is_date(ser.dtype.pyarrow_dtype):
+                        # GH#53854 to_pydatetime not supported for pyarrow date dtypes
+                        d = ser._values.astype(object)
```

Comment:
Could you use `ser._values.to_numpy(dtype=object)`?

Reply:
Just pushed a new commit with the requested changes.